### Utilizando a base de dados archive.ics.uci.edu/ml/datasets/Car+Evaluation:

In [3]:
import pandas as pd

data = pd.read_csv('car.csv')
data.head()

buying  maint doors persons lug_boot safety  class
0  vhigh  vhigh     2       2    small    low  unacc
1  vhigh  vhigh     2       2    small    med  unacc
2  vhigh  vhigh     2       2    small   high  unacc
3  vhigh  vhigh     2       2      med    low  unacc
4  vhigh  vhigh     2       2      med    med  unacc

### 1. (10 pontos) Defina o classificador cego com o menor erro de classificação possível. 

##### calcular a probabilidade a priori de cada classe

In [9]:
tot  = data['class'].size
tot

1728

In [10]:
cont = data['class'].value_counts()
cont

unacc    1210
acc       384
good       69
vgood      65
Name: class, dtype: int64

In [16]:
sum = 0
for i in cont:
    prob = i/tot
    print(prob)
    sum = sum + prob
    
sum

0.7002314814814815
0.2222222222222222
0.03993055555555555
0.03761574074074074


1.0

##### Erro maximo para cada classe

In [18]:
#unacc
1 - (cont[0]/tot)

0.2997685185185185

In [21]:
#acc
1 - (cont[1]/tot)

0.7777777777777778

In [22]:
#good
1 - (cont[2]/tot)

0.9600694444444444

In [25]:
#vgood
1 - (cont[3]/tot)

0.9623842592592593

<font color="yellow">Resposta: O classificador cego com o menor erro de classificação possível é unacc (Erro máximo = 0.2997685185185185)</font>

### 2. (18 pontos) Calcule as probablidades a priori para cada variável (para cada possível valor de cada variável).

### 3. (18 pontos) Calcule a probabilidade de cada valor de cada variável dado a classe (posteriori).

### 4. (18 pontos) Implemente o classificador Naive Bayes.

### 5. (18 pontos) Utilizando sua implementação e metade dos exemplos de cada classe como conjunto de treino e o resante como conjunto de teste, avalie o classificador Naive Bayes.

### 6. (18 pontos) Faça o mesmo da questão anterior utilizando uma implementação de biblioteca com configuração semelhente ao classificador solicitado na questão anterior.